blockchain/processing.py

```diff
@@ -216,6 +220,42 @@ def get_prior_hash(self, origin_id, phase):
 
         return prior_hash
 
+    def subscription_request(self, transaction):
+        """ check if given transaction has one or more subscriptions tied to it and inserts into subscriptions database
+            initial communication with subscription node
+        """
+        if "subscriptions" in transaction["payload"]:
+            subscriptions = transaction["payload"]['subscriptions']
+            for sub in subscriptions:
+                try:
+                    subscription = subscriptions[sub]
+                    subscription_id = str(uuid.uuid4())
+                    criteria = subscription['criteria']
+                    phase_criteria = subscription['phase_criteria']
+                    public_key = self.service_config['public_key']
+                    # store new subscription info
+                    sub_db.insert_subscription(subscription, subscription_id)
+                    # get subscription node
+                    subscription_node = self.network.get_subscription_node(subscription)
+                    # initial communication with subscription node
+                    if subscription_node:
+                        subscription_node.client.subscription_provisioning(subscription_id, criteria, phase_criteria, public_key)
```

Comment:
@Voxelot It looks like the implementation is in the network module.

Did you mean this call itself should be further abstracted?